src/main/java/dog/pawbook/logic/commands/AddDogCommand.java

```diff
@@ -53,6 +58,31 @@ protected String getDuplicateMessage() {
         return MESSAGE_DUPLICATE_DOG;
     }
 
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (model.hasEntity(toAdd)) {
+            throw new CommandException(getDuplicateMessage());
+        }
+
+        if (!model.hasEntity(toAdd.getOwnerId())) {
+            throw new CommandException(MESSAGE_OWNER_NOT_FOUND);
+        }
+
+        Owner o = (Owner) model.getFilteredEntityList().stream()
+            .filter(p -> p.getKey() == toAdd.getOwnerId())
+            .findFirst().orElseThrow()
+            .getValue();
+        int idEntity = model.addEntity(toAdd);
```

Comment:
Will it be better if we keep naming a little more consistent here? Since elsewhere in the I have already used something like `idEntityPair`, it will be confusing to see `idEntity` here since it can be ambiguous. I think a better alternative would be to simply name it `idNumber` or simply just `id`.